cfme/utils/template/base.py

```diff
@@ -256,6 +257,35 @@ def download_image(self):
                 logger.exception('Failed download of image using urllib')
                 return False
 
+        # Download checksum from url
+        checksum = None
```

Comment:
REQUIRED: please make it another method. this method is already big but your changes make it huge.

Reply:
@izapolsk Okay, so this new method should be called from download_image() or called separately from here https://github.com/ManageIQ/integration_tests/blob/master/cfme/utils/template/ec2.py#L69 ?

Reply:
@mmojzis, it depends on whether you want it to be run for all providers or ec2 only.
I'd rather you called it from run() since it would look more straightforward and clear.

Reply:
@izapolsk I would like to run it for template of any type.

Reply:
Although please take note that some templates like openshift ones may not have checksum files

Reply:
Yes, I can see it: http://file.cloudforms.lab.eng.rdu2.redhat.com/builds/cfme/5.10/stable/SHA256SUM
But actually template upload doesn't stop when checksum wasn't found:
https://github.com/ManageIQ/integration_tests/pull/8276/files#diff-c9ddd76cc9e9ceda813ce94622b76309R270

Reply:
but I'd rather change log level to warning in such case.

Reply:
Makes sense

Reply:
@mshriver What do you think about this? Where to put that method?

Reply:
The image is not downloaded for all provider upload types, so I'm not sure what our checksum validation case would be for when we didn't download the image (and it was directly uploaded).

Assuming this is only validating checksum of locally downloaded images, I think it can be called from `download_image` instead of from `run`.

Reply:
Created separate method that is called in download_image method. Also changed logger to warn when there is no SHA sum for downloaded image.
This can be tested when devs start building EC2 images.

Reply:
Forgot it can be tested with 5.10. So tested with ec2 and 5.10. Working well.